### DISCUSSION TITLE
Same typing with less boilerplate

### DISCUSSION THREAD

**Discussion by freddi301** [TanStack/router#637] [Ideas]
see condesandbox https://codesandbox.io/s/react-router-client-form-gd796z (for fully working types download the project and open in vsocde as create-react-app does not support typescript 5.x)

```typescript
createRouter({
  path: "",
  children: parent =>[
    route({ path: "user/$userId" })
  ]
});

navigate({
  path: "/user/$userId",
  params: { userId: "14" }
})
```

**Comment by tannerlinsley** [comment]
Can child routes ingest and use computed type information from their parent routes? With var types (as opposed to builder types) this is usually impossible….

**Reply by freddi301** [reply]
Can you explain what do you mean? What are var types vs builder types? Are there a scenrio when it's possible?
In the example the child routes do get type and runtime information about parent route path params.

**Reply by freddi301** [reply]
I edited the post to a more compelte example
